SimCalorimetry/EcalSimAlgos/src/EcalLiteDTUCoder.cc

```diff
@@ -0,0 +1,181 @@
+//When the gain 1 is selected we have:
+//5 samples before the saturated sample and 10 samples after.
+//When we have N consecutive saturated samples, the window at gain 1 is extended of N samples after (still to be implemented - to be double checked in the Cometti's PhD thesis)
+const float DTUVersion = 1.0;
+
+EcalLiteDTUCoder::EcalLiteDTUCoder(bool addNoise,
+                                   bool PreMix1,
+                                   EcalLiteDTUCoder::Noisifier* ebCorrNoise0,
+                                   EcalLiteDTUCoder::Noisifier* ebCorrNoise1)
+    : m_peds(nullptr),
+      m_gainRatios(nullptr),
+      m_intercals(nullptr),
+      m_maxEneEB(ecalPh2::maxEneEB),  // Maximum for CATIA: LSB gain 10: 0.048 MeV
+      m_addNoise(addNoise),
+      m_PreMix1(PreMix1)
+
+{
+  m_ebCorrNoise[0] = ebCorrNoise0;
+  m_ebCorrNoise[1] = ebCorrNoise1;
+}
+
+EcalLiteDTUCoder::~EcalLiteDTUCoder() {}
+
+void EcalLiteDTUCoder::setFullScaleEnergy(double EBscale) { m_maxEneEB = ecalPh2::maxEneEB; }
+
+void EcalLiteDTUCoder::setPedestals(const EcalLiteDTUPedestalsMap* pedestals) { m_peds = pedestals; }
+
+void EcalLiteDTUCoder::setGainRatios(const EcalCATIAGainRatios* gainRatios) { m_gainRatios = gainRatios; }
+
+void EcalLiteDTUCoder::setIntercalibConstants(const EcalIntercalibConstantsMC* ical) { m_intercals = ical; }
+
+double EcalLiteDTUCoder::fullScaleEnergy(const DetId& detId) const { return m_maxEneEB; }
+
+void EcalLiteDTUCoder::analogToDigital(CLHEP::HepRandomEngine* engine,
+                                       const EcalSamples& clf,
+                                       EcalDataFrame_Ph2& df) const {
+  df.setSize(clf.size());
+  encode(clf, df, engine);
+}
+
+void EcalLiteDTUCoder::encode(const EcalSamples& ecalSamples,
+                              EcalDataFrame_Ph2& df,
+                              CLHEP::HepRandomEngine* engine) const {
+  const unsigned int nSamples(ecalSamples.size());
+
+  DetId detId = ecalSamples.id();
+  double Emax = fullScaleEnergy(detId);
+
+  //N Gains set to 2 in the .h
+  double pedestals[ecalPh2::NGAINS];
+  double widths[ecalPh2::NGAINS];
+  double LSB[ecalPh2::NGAINS];
+  double trueRMS[ecalPh2::NGAINS];
+
+  double icalconst = 1.;
+  findIntercalibConstant(detId, icalconst);
+
+  for (unsigned int igain(0); igain < ecalPh2::NGAINS; ++igain) {
+    // fill in the pedestal and width
+    findPedestal(detId, igain, pedestals[igain], widths[igain]);
+    // insert an absolute value in the trueRMS
+    trueRMS[igain] = std::sqrt(std::abs(widths[igain] * widths[igain] - 1. / 12.));
+
+    LSB[igain] = Emax / (ecalPh2::MAXADC * ecalPh2::gains[igain]);
+  }
+
+  CaloSamples noiseframe[ecalPh2::NGAINS] = {
+      CaloSamples(detId, nSamples),
+      CaloSamples(detId, nSamples),
+  };
+
+  const Noisifier* noisy[ecalPh2::NGAINS] = {m_ebCorrNoise[0], m_ebCorrNoise[1]};
+
+  for (unsigned int ig; ig < ecalPh2::NGAINS; ++ig) {
```

Comment:
`unsigned int ig = 0`

Reply:
Right, i don't know why it was compiling locally. Done